sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -5,183 +5,116 @@
+} | {
+    status: "error";
+    error: Error | string;
+};
 
 // @public
-export interface Link {
-    attributes?: SpanAttributes;
-    context: SpanContext;
+export interface TracingClient {
+    createRequestHeaders(spanId: TracingSpanIdentifier): Record<string, string>;
+    parseTraceparentHeader(traceparentHeader: string): TracingSpanIdentifier | undefined;
+    startSpan<Options extends {
+        tracingOptions?: OperationTracingOptions;
+    }>(name: string, operationOptions?: Options, spanOptions?: TracingSpanOptions): {
+        span: TracingSpan;
+        tracingContext: TracingContext;
+        updatedOptions: Options;
+    };
+    withContext<CallbackArgs extends unknown[], Callback extends (...args: CallbackArgs) => ReturnType<Callback>>(context: TracingContext, callback: Callback, callbackThis?: ThisParameterType<Callback>, ...callbackArgs: CallbackArgs): ReturnType<Callback>;
+    withSpan<Options extends {
+        tracingOptions?: OperationTracingOptions;
+    }, Callback extends (updatedOptions: Options, span: Omit<TracingSpan, "end">) => ReturnType<Callback>>(name: string, callback: Callback, operationOptions?: Options, spanOptions?: TracingSpanOptions, callbackThis?: ThisParameterType<Callback>): Promise<ReturnType<Callback>>;
 }
 
 // @public
-export interface OperationTracingOptions {
-    tracingContext?: Context;
+export interface TracingClientOptions {
+    namespace: string;
+    packageInformation: {
+        name: string;
+        version?: string;
+    };
 }
 
 // @public
-export function setSpan(context: Context, span: Span): Context;
-
-// @public
-export function setSpanContext(context: Context, spanContext: SpanContext): Context;
-
-// @public
-export interface Span {
-    addEvent(name: string, attributesOrStartTime?: SpanAttributes | TimeInput, startTime?: TimeInput): this;
-    end(endTime?: TimeInput): void;
-    isRecording(): boolean;
-    recordException(exception: Exception, time?: TimeInput): void;
-    setAttribute(key: string, value: SpanAttributeValue): this;
-    setAttributes(attributes: SpanAttributes): this;
-    setStatus(status: SpanStatus): this;
-    spanContext(): SpanContext;
-    updateName(name: string): this;
+export interface TracingContext {
+    deleteValue(key: symbol): TracingContext;
+    getValue(key: symbol): unknown;
+    setValue(key: symbol, value: unknown): TracingContext;
 }
 
 // @public
-export interface SpanAttributes {
-    [attributeKey: string]: SpanAttributeValue | undefined;
+export interface TracingSpan {
+    end(): void;
+    isRecording(): boolean;
+    setAttribute(name: string, value: unknown): void;
+    setStatus(status: SpanStatus): void;
+    readonly tracingSpanId: TracingSpanIdentifier;
 }
 
 // @public
-export type SpanAttributeValue = string | number | boolean | Array<null | undefined | string> | Array<null | undefined | number> | Array<null | undefined | boolean>;
-
-// @public
-export interface SpanContext {
+export interface TracingSpanIdentifier {
```

Comment:
any reason for the naming? It seems to be based on `SpanContext`, can we  have a name that is similar (e.g. TracingSpanContext) to reduce cognitive load needed to map our APIs to otel?